executor/executor.go

```diff
@@ -1140,28 +1160,34 @@ func init() {
 			defer span1.Finish()
 			ctx = opentracing.ContextWithSpan(ctx, span1)
 		}
-
-		e := &executorBuilder{is: is, ctx: sctx}
-		exec := e.build(p)
-		if e.err != nil {
-			return nil, e.err
-		}
-		err := exec.Open(ctx)
-		defer terror.Call(exec.Close)
+		c := &execCtx{sctx: sctx, is: is, p: p}
+		_, err := c.buildExecutor()
 		if err != nil {
 			return nil, err
 		}
-		chk := newFirstChunk(exec)
-
-		err = Next(ctx, exec, chk)
+		err = c.exec.Open(ctx)
+		defer terror.Call(c.exec.Close)
 		if err != nil {
 			return nil, err
 		}
-		if chk.NumRows() == 0 {
-			return nil, nil
+		for {
```

Comment:
Does it execute the whole query inside `EvalSubqueryFirstRow`?

Reply:
IMO, it only execute subplan subquery for example: 

`select v from t where id = 1 for update` of `select (select v from t where id = 1 for update) v1, (select v from t where id = 2) from dual`